src/core_codemods/sql_parameterization.py

```diff
@@ -309,88 +399,74 @@ def leave_Module(self, original_node: cst.Module):
-        return True
+        t = _extract_prefix_raw_value(self, expression)
+        if not t:
+            return True
+        prefix, raw_value = t
+        if "b" in prefix:
+            return False
+        if "r" in prefix:
+            return raw_quote_pattern.fullmatch(raw_value) is None
+        return quote_pattern.fullmatch(raw_value) is None
+
+    def _is_injectable(self, expression: cst.BaseExpression) -> bool:
+        return not bool(BaseType.infer_expression_type(expression))
 
     def _is_literal_start(self, node: cst.CSTNode, modulo_2: int) -> bool:
-        # TODO limited for now, won't include cases like "name = 'username_" + name + "_tail'"
-        match node:
-            case cst.SimpleString():
-                prefix = node.prefix.lower()
-                if "b" in prefix:
-                    return False
-                if "r" in prefix:
-                    matches = list(self.raw_quote_pattern.finditer(node.raw_value))
-                else:
-                    matches = list(self.quote_pattern.finditer(node.raw_value))
-                # avoid cases like: "where name = 'foo\\\'s name'"
-                # don't count \\' as these are escaped in string literals
-                return (
-                    matches[-1].end() == len(node.raw_value)
-                    if matches and len(matches) % 2 == modulo_2
-                    else False
-                )
-            case cst.FormattedStringText():
-                # TODO may be in the middle i.e. f"name='home_{exp}'"
-                # be careful of f"name='literal'", it needs one but not two
-                return False
-        return False
+        t = _extract_prefix_raw_value(self, node)
+        if not t:
+            return False
+        prefix, raw_value = t
+        if "b" in prefix:
+            return False
+        if "r" in prefix:
+            matches = list(raw_quote_pattern.finditer(raw_value))
+        else:
+            matches = list(quote_pattern.finditer(raw_value))
+        # avoid cases like: "where name = 'foo\\\'s name'"
+        # don't count \\' as these are escaped in string literals
+        return (matches != None) and len(matches) % 2 == modulo_2
 
     def _is_literal_end(self, node: cst.CSTNode) -> bool:
-        match node:
-            case cst.SimpleString():
-                if "b" in node.prefix:
-                    return False
-                if "r" in node.prefix:
-                    matches = list(self.raw_quote_pattern.finditer(node.raw_value))
-                else:
-                    matches = list(self.quote_pattern.finditer(node.raw_value))
-                return matches[0].start() == 0 if matches else False
-            case cst.FormattedStringText():
-                # TODO may be in the middle i.e. f"'{exp}_home'"
-                # be careful of f"name='literal'", it needs one but not two
-                return False
-        return False
+        t = _extract_prefix_raw_value(self, node)
```

Comment:
Other than the final condition, this method exactly duplicates the body of `_is_literal_start`, which suggests it should be factored out.